### ISSUE TITLE
No authentication popup after a redirect

### ISSUE THREAD

**Issue by bertbsky** [MicrosoftEdge/WebView2Feedback#2273]
**Description**
Since CEF v96, authentication popups no longer appear after an initial redirect.
It does work fine after a page reload or back/forward navigation.

**Version**
SDK: 1.0.1150.38
Runtime: 99.0.1150.39
Framework: WPF
OS: Win10 21H1

**Repro Steps**
Set up a web server that redirects to a location that requires authentication.
For example:
* http://demo/
  -> server responds with 301 Moved Permanently and header Location: /auth/
* WebView2 follows the redirect to http://demo/auth/
  -> server responds with 401 Unauthorized and header WWW-Authenticate: Basic or WWW-Authenticate: Digest
* At this point the browser should display the authentication popup and send another request to http://demo/auth/ with an Authorization header, instead it stops and displays the 401 HTML content.

The same issue occured with CefSharp v96 where the GetAuthCredentials() method was not triggered but this has since been fixed in CefSharp v97. Edge WebView2 v93 was still working OK but is broken since v96.

Request/response flow:
```http
GET / HTTP/1.1
Host: demo
User-Agent: Chrome/99.0.4844.51
Accept: */*

HTTP/1.1 301 Moved Permanently
Server: lighttpd/1.4.50
Location: /auth/
Content-Length: 0
Content-Type: text/html
```
```http
GET /auth/ HTTP/1.1
Host: demo
User-Agent: Chrome/99.0.4844.51
Accept: */*

HTTP/1.1 401 Unauthorized
Server: lighttpd/1.4.50
WWW-Authenticate: Digest realm="restricted", charset="UTF-8", nonce="628c3094:24cfd6e8c7cf20b61b8431a443eefd94", qop="auth"
Content-Length: 381
Content-Type: text/html
```

[AB#38543568](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/38543568)

**Comment by champnic** [comment]
Thanks for the bug report @bertbsky - sounds like we'll have to do a similar fix as CEF. I've added this bug to our backlog to take a look. Thanks!

**Comment by yildirimcagri** [comment]
Hi @bertbsky, does this repro for you with the latest Edge Canary runtime?  I'm unable to reproduce this, so wanted to check with you. The other difference could be that my server is simulating HTTP basic authentication, but it sounded like you also saw repro with that. Thanks!

**Comment by bertbsky** [comment]
Hi @yildirimcagri, after some further testing in a simplified environment, I discovered that there is one additional requirement to make this issue appear: there needs to be a **WebResourceRequested** filter defined:

```csharp
web.CoreWebView2.AddWebResourceRequestedFilter("*", CoreWebView2WebResourceContext.All);
web.CoreWebView2.WebResourceRequested += CoreWebView2_WebResourceRequested;

private void CoreWebView2_WebResourceRequested(object sender, CoreWebView2WebResourceRequestedEventArgs e)
{
    // empty
}
```
It does still occur with all versions that I could test with: 96, 99, 100, 102 (canary).

**Comment by akielbas** [comment]
Today we root caused an NTLM authentication issue (no username / password prompt) to the same handler. Using the C++ API and adding even an empty `add_WebResourceRequested` handler results in no prompts. Removing the handler restores the proper prompt behavior. In our case the 401 followed a 302 redirect.

**Comment by akielbas** [comment]
Our users are able to workaround this issue by reloading the page (Ctrl + R), upon reload the proper prompt is shown.

**Comment by bertbsky** [comment]
I cannot say this works for me, as soon as I filter on Document type requests the popup is broken:
```
web.CoreWebView2.AddWebResourceRequestedFilter("*", CoreWebView2WebResourceContext.Document);
```
Funnily enough, adding all individual filters causes the event to stop working (and the auth popup to work again). I haven't checked which exact combination of filters breaks this.
```
web.CoreWebView2.AddWebResourceRequestedFilter("*", CoreWebView2WebResourceContext.Document);
web.CoreWebView2.AddWebResourceRequestedFilter("*", CoreWebView2WebResourceContext.Stylesheet);
web.CoreWebView2.AddWebResourceRequestedFilter("*", CoreWebView2WebResourceContext.Image);
web.CoreWebView2.AddWebResourceRequestedFilter("*", CoreWebView2WebResourceContext.Manifest);
web.CoreWebView2.AddWebResourceRequestedFilter("*", CoreWebView2WebResourceContext.Other);
```

**Comment by champnic** [comment]
This is fixed in runtime versions 107.0.1389.0+, and we are working on backporting to 106 versions. Thanks!

**Comment by monica-ch** [comment]
This is fixed in runtime versions 107.0.1415.0+. Thanks!
